src/auto/injector.js

```diff
@@ -826,8 +826,10 @@ function createInjector(modulesToLoad, strictDi) {
       if (msie <= 11) {
         return false;
       }
+      // Workaround for MS Edge.
+      // Check https://connect.microsoft.com/IE/Feedback/Details/2211653
       return typeof func === 'function'
-        && /^class\s/.test(Function.prototype.toString.call(func));
+        && /^(?:class\s|constructor\()/.test(Function.prototype.toString.call(func));
```

Comment:
Actually I think we need to test for `constructor(` only if we are on Edge, since it is possible that someone might have a function actually called `constructor`.